custom/icds_reports/tasks.py

```diff
@@ -123,70 +141,118 @@ def _update_aggregate_locations_tables(cursor):
-                            aggregation_type, settings.SERVER_ENVIRONMENT, aggregation_date
-                        )
-                    )
-                    notify_exception(
-                        None,
-                        message="Error occurred during ICDS aggregation",
-                        details={
-                            'type': aggregation_type,
-                            'date': aggregation_date,
-                            'error': exc,
-                        }
-                    )
-                    self.retry(exc=exc)
-
-                celery_task_logger.info(
-                    "Ended icds reports {} update_{}_aggregate_tables".format(
-                        aggregation_date, aggregation_type
-                    )
-                )
-
-    if future_tasks:
-        aggregate_tables.delay(future_tasks[0], future_tasks[1:])
-    else:
-        # temporary soft assert to verify it's completing
-        _dashboard_team_soft_assert(False, "Aggregation completed on {}".format(settings.SERVER_ENVIRONMENT))
-        celery_task_logger.info("Aggregation has completed")
-        icds_data_validation.delay(aggregation_date)
+
+def aggregate_awc_daily(day):
+    _run_custom_sql_script(["SELECT aggregate_awc_daily(%s)"], day)
+
+
+def _update_months_table(day):
+    _run_custom_sql_script(["SELECT update_months_table(%s)"], day)
+
+
+def _child_health_monthly_table(day):
+    _run_custom_sql_script([
+        "SELECT create_new_table_for_month('child_health_monthly', %s)",
+        "SELECT insert_into_child_health_monthly(%s)"
+    ], day)
+
+
+def _ccs_record_monthly_table(day):
+    _run_custom_sql_script([
+        "SELECT create_new_table_for_month('ccs_record_monthly', %s)",
+        "SELECT insert_into_ccs_record_monthly(%s)"
+    ], day)
+
+
+def _daily_attendance_table(day):
+    _run_custom_sql_script([
+        "SELECT create_new_table_for_month('daily_attendance', %s)",
+        "SELECT insert_into_daily_attendance(%s)"
+    ], day)
+
+
+def _agg_child_health_table(day):
+    _run_custom_sql_script([
+        "SELECT create_new_aggregate_table_for_month('agg_child_health', %s)",
+        "SELECT aggregate_child_health(%s)"
+    ], day)
+
+
+def _agg_ccs_record_table(day):
+    _run_custom_sql_script([
+        "SELECT create_new_aggregate_table_for_month('agg_ccs_record', %s)",
+        "SELECT aggregate_ccs_record(%s)"
+    ], day)
+
+
+def _agg_awc_table(day):
+    _run_custom_sql_script([
+        "SELECT create_new_aggregate_table_for_month('agg_awc', %s)",
+        "SELECT aggregate_awc_data(%s)"
+    ], day)
+
+
+@task(queue='background_queue')
+def email_dashboad_team(aggregation_date):
+    # temporary soft assert to verify it's completing
+    _dashboard_team_soft_assert(False, "Aggregation completed on {}".format(settings.SERVER_ENVIRONMENT))
+    celery_task_logger.info("Aggregation has completed")
+    icds_data_validation.delay(aggregation_date)
```

Comment:
`email_dashboard_team` makes it sound like a monitoring-only utility, but the `icds_data_validation` call looks important - does that perform sanity checks?

Reply:
Yeah. that provides sanity checks. I didn't catch this earlier, but it should get called in parallel with emailing the team. (I plan on removing this task anyways once we have some better tracking)